From d97614fd4f76bf38dea2565d1c630d7875008f9c Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:37:02 -0400
Subject: [scsi] enable destruction of blocked devices which fail LUN scanning

Message-id: <1402670238-13916-135-git-send-email-snitzer@redhat.com>
Patchwork-id: 83889
O-Subject: [RHEL7.1 PATCH 134/150] scsi: enable destruction of blocked devices which fail LUN scanning
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

BZ: 1105204

Upstream commit 0516c08d10835a8f9169051504cbc9929b3a7f3e
Author: Bart Van Assche <bvanassche@acm.org>
Date:   Tue Jul 2 15:06:33 2013 +0200

    [SCSI] enable destruction of blocked devices which fail LUN scanning

    If something goes wrong during LUN scanning, e.g. a transport layer
    failure occurs, then __scsi_remove_device() can get invoked by the
    LUN scanning code for a SCSI device in state SDEV_CREATED_BLOCK and
    before the SCSI device has been added to sysfs (is_visible == 0).
    Make sure that even in this case the transition into state SDEV_DEL
    occurs. This avoids that __scsi_remove_device() can get invoked a
    second time by scsi_forget_host() if this last function is invoked
    from another thread than the thread that performs LUN scanning.

    Signed-off-by: Bart Van Assche <bvanassche@acm.org>
    Signed-off-by: James Bottomley <JBottomley@Parallels.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_lib.c b/drivers/scsi/scsi_lib.c
index 10d1cec..4074edd 100644
--- a/drivers/scsi/scsi_lib.c
+++ b/drivers/scsi/scsi_lib.c
@@ -2223,6 +2223,7 @@ scsi_device_set_state(struct scsi_device *sdev, enum scsi_device_state state)
   case SDEV_OFFLINE:
   case SDEV_TRANSPORT_OFFLINE:
   case SDEV_CANCEL:
+  case SDEV_CREATED_BLOCK:
    break;
   default:
    goto illegal;
-- 
1.7.1